### PR TITLE
Arrays

### PR DESCRIPTION
**Pull request by r3ddawn**

### CODE REVIEW

File: test/ArrayTest.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import lists.Array;
+import static org.junit.Assert.*;
+import org.junit.Test;
+
+public class ArrayTest {
```
Comment (booleangate) [inline test/ArrayTest.java]: These tests look the same as the linked list tests. How can we reuse that code?

File: src/lists/Array.java [hunk 2]
```diff
@@ -1,31 +1,96 @@
...
+    @Override
     public boolean isEmpty() {
-    return false;     
+        return size == 0;     
     }
     
+    @Override
     public String get(int index){
-    return null;
+        return array[index];
```
Comment (booleangate) [inline src/lists/Array.java]: You need bounds checking here.

File: src/lists/Array.java [hunk 3]
```diff
@@ -1,31 +1,96 @@
...
-       
+        if (size == 0) {
+            array = new String[1];
+            array[0] = value;
+            size++;
+            return;
+        }
+        // New array to move values
+        String[] temp = new String[array.length];
+        temp = array;
```
Comment (booleangate) [inline src/lists/Array.java]: You can just do `String[] temp = array;`

File: src/lists/Array.java [hunk 4]
```diff
@@ -1,31 +1,96 @@
 package lists;
 
 public class Array implements List {
+    private int size;
```
Comment (booleangate) [inline src/lists/Array.java]: You don't need this variable since you can always get this information from `array.size`

File: src/lists/Array.java [hunk 5]
```diff
@@ -1,31 +1,96 @@
...
     
+    @Override
     public String get(int index){
-    return null;
+        return array[index];
     }
     
+    @Override
     public void add(String value, int index){
+        if (size == 0 && index == 0) {
```
Comment (booleangate) [inline src/lists/Array.java]: You need bounds checking here.

File: src/lists/Array.java [hunk 6]
```diff
@@ -1,31 +1,96 @@
...
+            array = new String[1];
+            array[index] = value;
+            size++;
+            return;
+        }
+        // New array to move values
+        String[] temp = new String[array.length];
+        temp = array;
```
Comment (booleangate) [inline src/lists/Array.java]: These two lines can be combined into  `String[] temp = array;`

File: src/lists/Array.java [hunk 7]
```diff
@@ -1,31 +1,96 @@
...
+        // Adding new value to the array
+        array[array.length - 1] = value;
+        size++;
     }
     
+    @Override
     public void delete(int index){
+        // New array to move values
+        String[] temp = new String[array.length];
+        temp = array;
```
Comment (booleangate) [inline src/lists/Array.java]: Combine

File: test/ArrayTest.java [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+    @Test
+    public void size() {
+        Array array = new Array();
+        
+        assertEquals(0, array.size());
```
Comment (booleangate) [inline test/ArrayTest.java]: Consider adding a test case for a non-empty array

File: test/ArrayTest.java [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+    }
+    
+    @Test
+    public void isEmpty() {
+        Array array = new Array();
+        
+        assertTrue(array.isEmpty());
```
Comment (booleangate) [inline test/ArrayTest.java]: Consider adding a test case for a non-empty array
